.github/workflows/ci.yml

```diff
@@ -105,14 +103,18 @@ jobs:
               uses: actions/setup-python@v5
               with:
                   python-version: ${{ matrix.python-version }}
-                  cache: pip
-                  cache-dependency-path: |
-                      **/setup.cfg
-                      **/pyproject.toml
-                      **/requirements*.txt
+
+            - name: Install uv
+              run: curl --proto '=https' --tlsv1.2 -LsSf https://github.com/astral-sh/uv/releases/download/0.1.21/uv-installer.sh | sh
 
             - name: Install package
-              run: pip install -e .[dev,smiles,optimade] aiida-core==${{ matrix.aiida-core-version }}
+            # NOTE: uv (unlike pip) does not compile python to bytecode after install.
+            # This uncovered a lot of SyntaxError(s) in the vapory package,
+            # since pip swallows these by default (WTH?).
+            # Ideally, these would be fixed, but vapory is largely unmaintained,
+            # so here we simply keep the pip behaviour with the --compile flag.
+            # See https://github.com/astral-sh/uv/issues/1928#issuecomment-1968857514
+              run: uv pip install --compile --system "aiidalab_widgets_base[dev,smiles,optimade] @ ." aiida-core==${{ matrix.aiida-core-version }}
```

Comment:
Note: `uv` currently does not allow the simpler syntax ".[dev,smiles,optimade]`. But it will in the future.

Reply:
I guess beside this, all other stuff will be the same? It is not another package manage but simply an installer I assume? Which means the installed packages can be inspected by the normal pip command, correct?

Reply:
Yes, `uv` is only an installer for now, for the most part you simply substitute `pip ...` with `uv pip ...`. There are difference in behaviour, as explained in `uv`s README, and not everything is implemented yet, but it should be stable enough for the CI to use (and we pin the version).

Reply:
>  Note: uv currently does not allow the simpler syntax .[dev,smiles,optimade]. But it will in the future.

This simpler syntax is actually allowed for editable installs, so I switched to editable install.